connector/connect/src/main/protobuf/spark/connect/relations.proto

```diff
@@ -213,7 +213,7 @@ message Deduplicate {
 
 message LocalRelation {
   repeated Expression.QualifiedAttribute attributes = 1;
```

Comment:
what about removing this field and also read the schema from the arrow batch?  @grundprinzip

Reply:
For short term it should work. We probably only need `name` and `type` to ask server side construct such attributes for the local relation.

For longer term I am not sure. Depending on if there are other extra information that LocalRelation needs from such attributes.

Reply:
each arrow_batch in collect starts with the schema, it will be consistent if we also do this in createDataFrame

Reply:
Sure I am not against to use arrow schema for now.

Reply:
I find we lack a `fromBatchWithSchemaIterator` method correspond to `toBatchWithSchemaIterator`, so I will implement one.